plugin/evm/header/extra.go

```diff
@@ -66,14 +71,40 @@ func VerifyExtra(rules extras.AvalancheRules, extra []byte) error {
 			)
 		}
 	default:
-		if uint64(extraLen) > params.MaximumExtraDataSize {
+		if uint64(extraLen) > MaximumExtraDataSize {
 			return fmt.Errorf(
 				"%w: expected <= %d but got %d",
 				errInvalidExtraLength,
-				params.MaximumExtraDataSize,
+				MaximumExtraDataSize,
 				extraLen,
 			)
 		}
 	}
 	return nil
 }
+
+// PredicateBytesFromExtra returns the predicate result bytes from the header's
+// extra data. If the extra data is not long enough, an empty slice is returned.
+func PredicateBytesFromExtra(extra []byte) []byte {
+	// Prior to Durango, the VM enforces the extra data is smaller than or equal
+	// to this size.
+	// After Durango, the VM pre-verifies the extra data past the dynamic fee
+	// rollup window is valid.
+	if len(extra) <= FeeWindowSize {
+		return nil
+	}
+	return extra[FeeWindowSize:]
+}
+
+// SetPredicateBytesInExtra sets the predicate result bytes in the header's extra
+// data. If the extra data is not long enough (i.e., an incomplete header.Extra
+// as built in the miner), it is padded with zeros.
+func SetPredicateBytesInExtra(extra []byte, predicateBytes []byte) []byte {
```

Comment:
this is a little jank but it's the cost of the way we're passing the predicate bytes into libevm. unless you can come up with a better idea I think it's fine for now.

Reply:
Fine with me!